---
author: "Andrew E. Jaffe and Kasper D. Hansen"
title: Comparison of Recount with GTEx"
---

# Overview

This document compares GTEx data release v6 to Recount.  The main issue addressed in this document is mapping up genes and samples between the two datasets.  The annotations are different:

- GTEx uses Gencode v19 mapped to hg19.
- Recount uses UCSC knownGene as represented by the `TxDb.Hsapiens.UCSC.hg38.knownGene` package, mapped to hg38.

# Dependencies

## R packages

```{r dependencies}
library(SummarizedExperiment)
library(stringr)
library(recount)
library(rtracklayer)
library(org.Hs.eg.db)
library(ballgown)
library(coop)
library(readr)
```

## Data objects

### From Recount

### From GTEx website

We have downloaded the annotation GTF files as well as the raw gene count matrix from the [GTEx portal](http://www.gtexportal.org).

```{r gtexWebsite}
dataPath="/dcs01/ajaffe/GTEX/V6/" # wherever data was downloaded
gtexGtf <- import(paste0(dataPath,"/gencode.v19.genes.patched_contigs.gtf"))
gtexData <- read_tsv(paste0(dataPath,"GTEx_Analysis_v6_RNA-seq_RNA-SeQCv1.1.8_gene_reads.gct.gz"), skip=2)
gtexCounts <- gtexData[,3:ncol(gtexData)]
rownames(gtexCounts) <- gtexData$Name
```

### From elsewhere

These are the the Rail-RNA processed samples

```{r loadObjectsRecount}
load("/dcl01/leek/data/recount-website/rse/rse_gtex/SRP012682/rse_gene.Rdata")
gtexPd = colData(rse_gene)
```

Let's match everything up.

```{r matchIDs}
mm = match(colnames(gtexCounts), gtexPd$sampid)
gtexCounts = gtexCounts[,!is.na(mm)]
gtexPd = gtexPd[mm[!is.na(mm)],]
rse_gene = rse_gene[,mm[!is.na(mm)]]
```

# Mapping GTEx annotation

We map between version by using ENTREZ gene ids.  The Recount representation is already using ENTREZ gene ids, but we need to map GTEx data to ENTREZ.

```{r GTExmap}
gtexMap <- gtexGtf[!duplicated(gtexGtf$gene_id)]
names(gtexMap) <- gtexMap$gene_id
gtexMap <- gtexMap[rownames(gtexCounts)]
stopifnot(all(rownames(gtexMap) == rownames(gtexCounts)))
gtexMap$EnsemblGeneID <- ballgown:::ss(gtexMap$gene_id, "\\.")
eid2ens = select(org.Hs.eg.db,gtexMap$EnsemblGeneID,
                "ENTREZID", "ENSEMBL")
eid2ens = CharacterList(split(eid2ens$ENTREZID, eid2ens$ENSEMBL))
gtexMap$EntrezID = eid2ens[gtexMap$EnsemblGeneID]
table(elementLengths(gtexMap$EntrezID))
table(gtexMap$gene_type,elementLengths(gtexMap$EntrezID) == 1)
```

Basically, only protein coding genes have a ENTREZ id.  We keep only the protein coding genes which are uniquely mapped to an ENTREZ id.

```{r keepMapping}
eidIndex <- which(elementLengths(gtexMap$EntrezID)==1 & gtexMap$gene_type == "protein_coding")
gtexMap <- gtexMap[eidIndex,]
gtexCounts <- gtexCounts[eidIndex,]
gtexMap$EntrezID <- sapply(gtexMap$EntrezID,"[", 1)
```

Now, we still have multiple Ensembl gene ids mapping to the same ENTREZ id.  We drop those as well

```{r dropMultiEntrez}
dropIDs <- gtexMap$EntrezID[duplicated(gtexMap$EntrezID)]
keepIdx <- which(!gtexMap$EntrezID %in% dropIDs)
gtexCounts <- gtexCounts[keepIdx,]
gtexMap <- gtexMap[keepIdx,]
rownames(gtexCounts) <- names(gtexMap) <- gtexMap$EntrezID
dim(gtexCounts)
```


Let's load data from Recount.

```{r}
rse_gene_scale <- scale_counts(rse_gene)
recountCounts = assays(rse_gene_scale)$counts
recountMap = rowRanges(rse_gene_scale)
stopifnot(all(colnames(recountCounts) == rownames(gtexPd)))

## match by gene
geneMatch = match(recountMap$gene_id, gtexMap$EntrezID )
recountCounts = recountCounts[!is.na(geneMatch),]
recountMap = recountMap[!is.na(geneMatch)]

gtexMap = gtexMap[geneMatch[!is.na(geneMatch)],]
gtexCounts = gtexCounts[geneMatch[!is.na(geneMatch)],]
stopifnot(all(rownames(gtexCounts)== rownames(recountCounts)))

save(gtexCounts, gtexMap, gtexPd, 
     recountCounts, recountMap,
     file = "/dcl01/leek/data/recount-website/GTEx_checks/GTEx_websiteAndRecount_EntrezMatched_17559_vs_8551_reproduced.rda",
     compress=TRUE)
```

# Comparison

```{r loadMap}
load("/dcl01/leek/data/recount-website/GTEx_checks/GTEx_websiteAndRecount_EntrezMatched_17559_vs_8551.rda")
gtexCounts <- as.matrix(gtexCounts)
```

```{r rawCounts comparisons}
rawCors <- sapply(1:nrow(gtexCounts), function(ii) pcor(gtexCounts[ii,], recountCounts[ii,]))
summary(rawCors)
sum(rawCors<=0.95, na.rm = TRUE)
sum(rawCors<=0.80, na.rm = TRUE)
mean(rawCors >= 0.99, na.rm = TRUE)
```

```{r rawCountsPlot, plot=TRUE}
pdf(file = "rawCorrelations.pdf")
dens <- density(rawCors, from = -1, to = 1, na.rm = TRUE, n = 4096)
plot(dens, xlab = "Pearson correlation", main = "Raw counts")
plot(dens, xlab = "Pearson correlation",
     main = "Raw counts", xlim = c(0.9,1))
```

```{r normCounts}
gtexCounts2 <- log2(sweep(gtexCounts, MARGIN = 2, FUN = "/", colSums(gtexCounts)/(4*10^7)) + 1)
recountCounts2 <- log2(assay(rse_gene2, "counts")[rownames(recountCounts), colnames(recountCounts)] + 1)
normCors <- sapply(1:nrow(gtexCounts2), function(ii) pcor(gtexCounts2[ii,], recountCounts2[ii,]))
summary(normCors)
```

```{r normCountsPlot, plot=TRUE}
pdf(file = "normCors.pdf")
plot(density(normCors, from = -1, to = 1, na.rm = TRUE),
     xlab = "Pearson correlation", main = "Gene counts, normalized for library size")
dev.off()
```

# Differential expression

Between colon and blood

```{r}
gtex_metadata <- all_metadata("gtex")
gtex_metadata2 <- subset(gtex_metadata, smtsd %in% c("Whole Blood", "Colon"))[, c("run", "smtsd", "smgebtch")]
colnames(gtex_metadata2)[c(2,3)] <- c("tissue", "batch")
gtex_metadata2 <- subset(gtex_metadata2, run %in% colnames(recountCounts2))
rownames(gtex_metadata2) <- gtex_metadata2$run
recountCounts2.de <- recountCounts2[, gtex_metadata2$run]
gtexCounts2.de <- gtexCounts2[, gtex_metadata2$run]

## need to get correct counts
design <- model.matrix(~ tissue + batch, levels = gtex_metadata2)


design <- gtex = model.matrix(~colData(gtex <- both)$tissue + colData(gtex <- both)$batch)
dge <- gtex = DGEList(counts=gtex <- both <- counts)
dge <- gtex = calcNormFactors(dge <- gtex)
v <- gtex <- voom(dge <- gtex,design <- gtex,plot=TRUE)
fit <- gtex <- lmFit(v <- gtex,design <- gtex)
fit <- gtex <- eBayes(fit <- gtex)
hist(fit <- gtex$p.value[,2],col=trop[2],main="DE",breaks=100)
```
