library('shiny')
library('shinyBS')

shinyUI(navbarPage(
    title = 'recount2: analysis-ready RNA-seq gene and exon counts datasets',
    tabPanel('Datasets',
        tags$head(
            includeScript("google-analytics.js"),
            tags$script(src = 'highlight.pack.js'),
            tags$script(src = 'shiny-showcase.js'),
            tags$link(rel = "stylesheet", type = "text/css",
                href = "rstudio.css")
        ),
        bsAlert('updatealert'),
        HTML('<a href="https://github.com/leekgroup/recount-website" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'github-recount-web\', 1)">'), img(src='https://raw.githubusercontent.com/leekgroup/recount-website/master/website/recount.png', align = "left", 'style' = 'padding-right:18px', width = '300'), HTML('</a>'),
        h3('A multi-experiment resource of analysis-ready RNA-seq gene and exon count datasets'),
        p(strong('recount2'), ' is an online resource consisting of RNA-seq gene and exon counts as well as coverage bigWig files for 2041 different studies. It is the second generation of the ', HTML('<a href="http://bowtie-bio.sourceforge.net/recount/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'recount-old\', 1)">ReCount project</a>.'), ' The raw sequencing data were processed with ', HTML('<a href="http://rail.bio" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'rail-rna\', 1)">Rail-RNA</a>'), ' as described in the recount2 paper and at ', HTML('<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1118-6" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'paper-rail\', 1)">Nellore et al, Genome Biology, 2016</a>'), ' which created the coverage bigWig files. For ease of statistical analysis, for each study we created count tables at the gene and exon levels and extracted phenotype data, which we provide in their raw formats as well as in RangedSummarizedExperiment R objects (described in the ', HTML('<a href="http://bioconductor.org/packages/SummarizedExperiment" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-SE\', 1)">SummarizedExperiment</a>'), ' Bioconductor package). We also computed the mean coverage per study and provide it in a bigWig file, which can be used with the ', HTML('<a href="http://bioconductor.org/packages/derfinder" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-derfinder\', 1)">derfinder</a>'), ' Bioconductor package to perform annotation-agnostic differential expression analysis at the expressed regions-level as described at ', HTML('<a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkw852" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'paper-derfinder\', 1)">Collado-Torres et al, Nucleic Acids Research, 2017</a>.'), ' The count tables, RangedSummarizeExperiment objects, phenotype tables, sample bigWigs, mean bigWigs, and file information tables are ready to use and freely available here. We also created the ',  HTML('<a href="http://bioconductor.org/packages/recount" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-recount\', 1)">recount</a>'), ' Bioconductor package which allows you to search and download the data for a specific study. By taking care of several preprocessing steps and combining many datasets into one easily-accessible website, we make finding and analyzing RNA-seq data considerably more straightforward.'),
        tags$hr(),
        h3('Main publication'),
        tags$ul(
            tags$li(HTML('<b>Collado-Torres L</b>, <b>Nellore A</b>, Kammers K, Ellis SE, Taub MA, Hansen KD, Jaffe AE, Langmead B, Leek JT. <a href="http://www.nature.com/nbt/journal/v35/n4/full/nbt.3838.html" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'recount-nbiotech\', 1)">Reproducible RNA-seq analysis using <i>recount2</i></a>. <i>Nature Biotechnology</i>, 2017. doi: 10.1038/nbt.3838.'))
        ),
        tags$hr(),
        h3('Related publications'),
        tags$ul(
            tags$li(HTML('<b>Nellore A</b>, Collado-Torres L, Jaffe AE, Alquicira-Hernández J, Wilks C, Pritt J, Morton J, Leek JT, Langmead B. <a href="https://academic.oup.com/bioinformatics/article/33/24/4033/2525684" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'rail-rna-paper\', 1)">Rail-RNA: scalable analysis of RNA-seq splicing and coverage</a>. <i>Bioinformatics</i>, 2017. doi: 10.1093/bioinformatics/btw575.')),
            tags$li(HTML('<b>Nellore A</b>, Wilks C, Hansen KD, Leek JT, Langmead B. <a href="https://academic.oup.com/bioinformatics/article/32/16/2551/1743271" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'rail-rna-dbgap-paper\', 1)">Rail-dbGaP: analyzing dbGaP-protected data in the cloud with Amazon Elastic MapReduce</a>. <i>Bioinformatics</i>, 2016. doi: 10.1093/bioinformatics/btw177.')),
            tags$li(HTML('<b>Nellore A</b>, Jaffe AE, Fortin JP, Alquicira-Hernández J, Collado-Torres L, Wang S, Phillips RA, Karbhari N, Hansen KD, Langmead B, Leek JT. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-1118-6" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'rail-rna-paper\', 1)">Human splicing diversity and the extent of unannotated splice junctions across human RNA-seq samples on the Sequence Read Archive</a>. <i>Genome Biology</i>, 2016. doi: 10.1186/s13059-016-1118-6.')),
            tags$li(HTML('<b>Ellis SE</b>, Collado-Torres L, Jaffe AE, Leek JT. <a href="https://doi.org/10.1093/nar/gky102" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'phenopredict\', 1)">Improving the value of public RNA-seq expression data by phenotype prediction</a>. <i>Nucleic Acids Research</i>, 2018. doi: 10.1093/nar/gky102.')),
            tags$li(HTML('<b>Collado-Torres L</b>, Nellore A, Jaffe AE. <a href="https://f1000research.com/articles/6-1558/v1" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'recount-workflow\', 1)">recount workflow: Accessing over 70,000 human RNA-seq samples with Bioconductor</a>. <i>F1000Research</i>, 2017. doi: 10.12688/f1000research.12223.1.')),
            tags$li(HTML('<b>Wilks C</b>, Gaddipati P, Nellore A, Langmead B. <a href="https://academic.oup.com/bioinformatics/article/34/1/114/4101942" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'snaptron-bioinfo\', 1)">Snaptron: querying splicing patterns across tens of thousands of RNA-seq samples</a>. <i>Bioinformatics</i>, 2018. doi: 10.1093/bioinformatics/btx547.')),
            tags$li(HTML('<b>Fu J</b>, Kammers K, Nellore A, Collado-Torres L, Leek JT, Taub MA. <a href="https://www.biorxiv.org/content/early/2018/05/25/247346" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'tx-biorxiv\', 1)">RNA-seq transcript quantification from reduced-representation data in recount2</a>. <i>bioRxiv</i>, 2018. doi: 10.1101/247346.')),
            tags$li(HTML('<b>Kuri-Magaña H</b>, Collado-Torres L, Jaffe AE, Valdovinos-Torres H, Ovilla-Muñoz M, Téllez-Sosa JM, Bonifaz-Alfonzo LC, Martínez-Barnetche J. <a href="https://www.frontiersin.org/articles/10.3389/fimmu.2018.02679/full" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'ncCSR-frontimmmuno\', 1)">Non-coding Class Switch Recombination-Related Transcription in Human Normal and Pathological Immune Responses</a>. <i>Front. Immunol.</i>, 2018. doi: 10.3389/fimmu.2018.02679.')),
            tags$li(HTML('<b>Ling JP</b>, Wilks C, Charles R, Ghosh D, Liang L, Santiago CP, Pang B, Venkataraman A, Clark BS, Nellore A, Langmead B, Blackshaw S. <a href="https://www.biorxiv.org/content/10.1101/501882v1" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'ascot-biorxiv\', 1)">ASCOT identifies key regulators of neuronal subtype-specific splicing</a>. <i>bioRxiv</i>, 2018. doi: 10.1101/384172.')),
            tags$li(HTML('<b>Madugundu AK</b>, Na CH, Nirujogi RS, Renuse S, Kim KP, Burns KH, Wilks C, Langmead B, Ellis SE, Collado-Torres L, Halushka MK, Kim M-S, Pandey A. <a href="https://onlinelibrary.wiley.com/doi/10.1002/pmic.201800315" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'proteomics-recount\', 1)">Integrated transcriptomic and proteomic analysis of primary human umbilical vein endothelial cells</a>. <i>Proteomics</i>, 2019. doi: 10.1002/pmic.201800315.')),
            tags$li(HTML('<b>Razmara A</b>, Ellis SE, Sokolowski DJ, Davis S, Wilson MD, Leek JT, Jaffe AE, Collado-Torres L. <a href="https://www.biorxiv.org/content/10.1101/618025v1" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'recountbrain-biorxiv\', 1)">recount-brain: a curated repository of human brain RNA-seq datasets metadata</a>. <i>bioRxiv</i>, 2019. doi: 10.1101/618025.')),
            tags$li(HTML('<b>Imada EL</b>, <b>Sanchez DF</b>, Collado-Torres L, Wilks C, Matam T, Dinalankara W, Stupnikov A, Lobo-Pereira F, Yip CW, Yasuzawa K, Kondo N, Itoh M, Suzuki H, Kasukawa T, Hon CC, de Hoon H, Shin JW, Carninci P, FANTOM consortium, Jaffe AE, Leek JT, Favorov A, Franco GR, Langmead B, Marchionni L. <a href="https://www.biorxiv.org/content/10.1101/659490v1" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'fc-biorxiv\', 1)">Recounting the FANTOM Cage Associated Transcriptome</a>. <i>bioRxiv</i>, 2019. doi: 10.1101/659490.'))
        ),
        tags$hr(),
        h3('The Datasets'),
        DT::dataTableOutput('metadata'),
        tags$hr(),
        downloadButton('downloadData', 'Download list of studies matching search results'),
        helpText('Note that GTEx is separated from this list.'),
        tags$hr(),
        h3('Authors'),
        tags$ul(
            tags$li(HTML('<a href="http://lcolladotor.github.io/about.html" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-leo\', 1)">Leonardo Collado-Torres</a>')),
            tags$li(HTML('<a href="http://nellore.bio/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-abhi\', 1)">Abhinav Nellore</a>')),
			tags$li(HTML('<a href="http://www.biostat.jhsph.edu/~kkammers/index.html" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-kai\', 1)">Kai Kammers</a>')),
			tags$li(HTML('<a href="http://jtleek.com/people/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-shannon\', 1)">Shannon E Ellis</a>')),
			tags$li(HTML('<a href="http://biostat.jhsph.edu/~mtaub/About.html" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-margaret\', 1)">Margaret A Taub</a>')),
			tags$li(HTML('<a href="http://www.hansenlab.org/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-kasper\', 1)">Kasper D Hansen</a>')),
			tags$li(HTML('<a href="http://aejaffe.com/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-andrew\', 1)">Andrew E Jaffe</a>')),
            tags$li(HTML('<a href="http://www.cs.jhu.edu/~langmea/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-ben\', 1)">Ben Langmead</a>')),
            tags$li(HTML('<a href="http://jtleek.com/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'about-jeff\', 1)">Jeff Leek</a>'))
        ),
        tags$hr(),
        h3('Support'),
        p('If you need help with the ', HTML('<a href="http://bioconductor.org/packages/recount" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-recount\', 1)">recount</a>'), ' Bioconductor package please get in touch via the ', HTML('<a href="https://support.bioconductor.org/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-support\', 1)">Bioconductor support website</a> '), '(remember to use the ', strong('recount'), ' tag). ', HTML('Please check this post on <a href="http://lcolladotor.github.io/2017/03/06/How-to-ask-for-help-for-Bioconductor-packages/#.WMGjvBIrJoM">how to ask for help for Bioconductor packages</a>.'), ' For support on reproducing the recount2 project, please get in touch via the ', HTML('<a href="https://github.com/leekgroup/recount-website/issues" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'issues-recount-web\', 1)">recount-website repository</a>.'))
    ),
    tabPanel('Popular datasets',
        DT::dataTableOutput('popular')
    ),
    tabPanel('GTEx',
        p('This tab shows the information for the GTEx project. Due to its size, we also provide ranged summarized experiment objects (RSE) by tissue at the gene and exon levels.'),
        DT::dataTableOutput('gtex')
    ),
    tabPanel('TCGA',
        p('This tab shows the information for the TCGA project. Due to its size, we also provide ranged summarized experiment objects (RSE) by tissue at the gene and exon levels.'),
        DT::dataTableOutput('tcga')
    ),
    tabPanel('Documentation',
        h3('recount2 objects documentation'),
        p('All columns of the table below are sortable and searchable. The columns are as follows:'),
        h5('accession'),
        p('The SRA accession identifier for the study. The link points to SRA for a full description of the study.'),
        h5('number of samples'),
        p('The total number of samples available for the given study. Note that in some exceptional cases not all samples for a given study were analyzed with ', HTML('<a href="http://rail.bio" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'rail-rna\', 1)">Rail-RNA</a>.')),
        h5('species'),
        p('The species of the samples under study.'),
        h5('abstract'),
        p('The abstract describing the study as provided by SRA via the ', HTML('<a href="http://bioconductor.org/packages/SRAdb" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-SRAdb\', 1)">SRAdb</a>'), ' Bioconductor package.'),
        h5('RSE gene'),
        p('The RangedSummarizedExperiment object for the counts summarized at the gene level using the Gencode v25 (GRCh38.p7, CHR) annotation as provided by ', HTML('<a href="https://www.gencodegenes.org/releases/25.html" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'gencode\', 1)">Gencode</a>. Note that the GRanges object recount::recount_genes includes bp_length as a metadata column which is the sum of the exon widths. If two exons are overlapping, the overlapped bases are only counted once.')),
        h5('counts gene'),
        p('A tsv file with the count matrix used to create the RangedSummarizedExperiment object at the gene level. Version 2 files include the gene ids in an extra column.'),
        h5('RSE exon'),
        p('The RangedSummarizedExperiment object for the counts summarized at the exon level using the Gencode v25 (GRCh38.p7, CHR) annotation as provided by ', HTML('<a href="https://www.gencodegenes.org/releases/25.html" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'gencode\', 1)">Gencode</a>. This GRangesList object recount::recount_exons has 1 element per gene. For each gene this object contains the reduced exons (version 1) or disjoint exons (version 2) such that they are non-overlapping within a gene. See the version section below for more information.')),
        h5('counts exon'),
        p('A tsv file with the count matrix used to create the RangedSummarizedExperiment object at the exon level.'),
        h5('RSE junctions'),
        p('The RangedSummarizedExperiment object for the counts summarized at the junction level. This GRanges object has 1 element per junction. For each junction this object contains the transcript names for the Gencode v24 junctions, transcript names and gene ids matching those used in the RSE objects at the gene or exon levels (based on Gencode v25 CHR regions), junction class, proposed gene ids and symbols. The junction ids match those used in the jx_cov and jx_bed files explained below. This file is present only if the project has at least one junction detected.'),
        h5('counts jx'),
        p('A tsv file with the count matrix used to create the RangedSummarizedExperiment object at the junction level. This file is present only if the project has at least one junction detected.'),
        h5('RSE transcript'),
        p('A RangedSummarizedExperiment object with the transcript quantifications as done by Fu et al, bioRxiv, 2018.'),
        h5('Junction raw coverage file'),
        HTML('The link jx_cov points to the raw junction coverage file that contains the junction ids (comma-separated), the sample ids (comma-separated), and the actual coverage value for the junction. Sample ids can be matched to run and project accession numbers using <a href="https://jhubiostatistics.shinyapps.io/recount/sample_ids.tsv" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'sample-ids\', 1)">sample_ids.tsv</a> (junction id, project accession, run accession).'),
        h5('Junction BED file'),
        p('The link jx_bed points to the BED file with one entry per junction present in the given project. The name of the junction includes the junction id, the donor, acceptor, and overall (if present in both) transcript names based on Gencode v24. See the known issues for an important detail on these BED files.'),
        h5('phenotype'),
        p('The phenotype information (sample metadata) in a tsv file used for both RangedSummarizedExperiment objects. The table includes the SRA study id, the SRA sample id, the SRA experiment id, the SRA run id, the reads counts as reported by SRA, the number of reads aligned by Rail-RNA, the proportion of reads reported by SRA that aligned, whether the sample was paired-end or not, whether we think SRA misreported the paired-end label, the number of mapped read count by Rail-RNA, the coverage AUC, the ', HTML('<a href="http://www.cs.cmu.edu/~ckingsf/sharq/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'sharq\', 1)">SHARQ prototype</a>'), ' tissue, the ', HTML('<a href="http://www.cs.cmu.edu/~ckingsf/sharq/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'sharq\', 1)">SHARQ prototype</a>'), ' cell type, the biosample sumbmission date, the biosample publication date, the biosample update date, the average read length, the GEO accession id, the sample title as extracted from GEO, the sample characteristics as extracted from GEO, and the name of the coverage bigWig file.'),
        h5('files info'),
        p('A tsv file with the names of the uploaded files, the md5sum, the size in bytes and the url to download the actual files. Version 2 includes the md5sum for the transcript files.'),
        h5('Annotation used'),
        p('The Gencode v25 GFF3 file with the comprehensive gene annotation (CHR regions) is available at the ', HTML('<a href="https://www.gencodegenes.org/releases/25.html" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'gencode\', 1)">Gencode website</a>.')),
        h5('FANTOM-CAT/recount2 RSE file'),
        p('RangedSummarizedExperiment object with the FANTOM-CAT/recount2 annotation counts. See', HTML('<a href="https://www.biorxiv.org/content/10.1101/659490v1" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'fc-biorxiv\', 1)">Imada, Sanchez et al, bioRxiv, 2019</a>'), ' for more information on how this annotation was determined. The rowRanges() slot is a GRangesList with one element per gene, and then exons coordinates for that given gene. For the md5sum and file size information for these files, please check the ', HTML('<a href="https://github.com/leekgroup/recount-website/blob/master/fileinfo/fc_rc_files_info.tsv" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'fc-biorxiv\', 1)">fc_rc_files_info.tsv</a>'), ' file.'),
        tags$hr(),
        h3('Getting started with RangedSummarizedExperiment objects'),
        p('Check out the ', HTML('<a href="http://bioconductor.org/packages/recount" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-recount\', 1)">recount</a>'), ' Bioconductor package for how to download data from the recount2 project and get started with your own analysis. The ', HTML('<a href="http://leekgroup.github.io/recount/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'vignette-recount\', 1)">quick start vignette</a>'), ' is particularly helpful. For more details check the ', HTML('<a href="http://bioconductor.org/packages/SummarizedExperiment" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-SE\', 1)">SummarizedExperiment</a>'), ' Bioconductor package for an overview on RangedSummarizedExperiment objects. Most RNA-seq differential expression Bioconductor packages use them. In particular, the vignette of the ', HTML('<a href="http://bioconductor.org/packages/DESeq2" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-DESeq2\', 1)">DESeq2</a>'), ' Bioconductor package shows in detail how to perform an analysis with them. You might also want to use the ', HTML('<a href="http://bioconductor.org/packages/DEFormats" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-DEFormats\', 1)">DEFormats</a>'), ' Bioconductor package for converting these objects to other formats.'),
        tags$hr(),
        h3('Difference between versions'),
        h5('Version 1: reduced exons'),
        p('Exon counts are derived from reduced exons, such that each exonic base is only counted once.'),
        h5('Version 2: disjoint exons, released January 12, 2018'),
        p('Exon counts are derived from disjoint exons, which also result in each exonic based being counted just once. However, disjoint exons are more useful than reduced exons because it is possible to reconstruct the actual exons with these counts. The following code might be helpful to understand the difference.'),
        h5('Version 2: transcript RSE files, May, 2018'),
        p('Transcript RSE files were re-calculated by Fu et al. For details check the second version of the pre-print.'),
        includeMarkdown('reducedexons.md'),
        p('We realized this in the peer-review process of the recount workflow paper. For backward compatibility, we provide both versions of files. The disjoint exon count files are larger than the reduced exon counts. The gene counts did not change between versions, just a handful of gene symbols, for which we updated the gene files. The gene count text files (counts_gene.tsv.gz) now include the gene ids as an extra column as was requested by several users. The file information document now includes the md5sum for the new files (including the transcript files).'),
        tags$hr(),
        h3('Code used'),
        HTML('The code for analyzing the <a href="http://rail.bio" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'rail-rna\', 1)">Rail-RNA</a> output and creating this website is available via <a href="https://github.com/leekgroup/recount-website" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'github-recount-web\', 1)">GitHub</a>.'),
        tags$hr(),
        h3('Known issues'),
        tags$li('Rail-RNA run on SRA data'),
        HTML('Check the <a href="https://github.com/nellore/runs/blob/master/sra/v2/hg38/NOTES">NOTES</a> on the Rail-RNA run on SRA data (version 2) which describes why some samples were discarded. Also check this <a href="https://github.com/leekgroup/recount-website/issues/11">known issue</a> listing the samples with 0 reads downloaded that are then missing from the RSE files.'),
        tags$li('Single-cell studies with pooled samples'),
        HTML('Some single-cell studies like SRP058046 sequenced more than one single-cell but are available only as a pooled sample from the Sequence Read Archive. Currently these studies might not be useful for differential expression analysis via recount2 but can be used for checking if a given exon-exon junction is present or if a given region is expressed. Reported by <a href="https://www.helmholtz-muenchen.de/icb/">Lukas Simon</a>.'),
        tags$li('BED files have the right end off by 1 base pair'),
        HTML('Check the R code we used to read in these BED files <a href="https://github.com/leekgroup/recount-website/blob/master/rse/create_rse.R#L341-L346">here</a>. The coordinates in these files are end of exon1 + 1, start of exon2 -1 in zero-based coordinates and should have been end of exon1 + 1, start of exon 2 in zero-based coordinates to be a <a href="https://genome.ucsc.edu/FAQ/FAQformat#format1">proper BED file</a>.'),
        h5('Found an issue?'),
        HTML('If you found an issue with recount2, please email us or describe the problem at the <a href="https://github.com/leekgroup/recount-website/issues">recount-website issue tracker</a>. Thank you!'),
        h3('FAQ'),
        tags$li('When is the next version going to be released? We are currently (summer 2018) determining what will be the next phase of the recount project. Please stay tuned, thanks!')
    ),
    tabPanel('Download data with R',
        p('The following R code shows how to use the', HTML('<a href="http://bioconductor.org/packages/recount" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'bioc-recount\', 1)">recount</a>'), 'Bioconductor package for downloading data. In this example we will download the data for study SRP009615.'),
        includeMarkdown('download.md')
    ),
    tabPanel('Accessing recount2 via SciServer',
        HTML('Not everyone has over 8 terabytes of disk space available to download all the data from the recount2 project. However, thanks to <a href="http://www.sciserver.org/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'idies\', 1)">SciServer</a> you can access it locally via a Jupyter Notebook. If you do so and want to share your work, please let the <a href="http://www.sciserver.org/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'idies\', 1)">SciServer</a> maintainers know via Twitter at <a href="https://twitter.com/IDIESJHU" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'twitter-idies\', 1)">IDIESJHU</a>. The step by step instructions describing how to access recount2 via <a href="http://www.sciserver.org/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'idies\', 1)">SciServer</a> are available in the <a href="http://bioconductor.org/packages/release/bioc/vignettes/recount/inst/doc/recount-quickstart.html#accessing-recount-via-sciserver" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'vignette-recount\', 1)">recount vignette</a>.'),
        p(HTML('<a href="http://bioconductor.org/packages/release/bioc/vignettes/recount/inst/doc/recount-quickstart.html#accessing-recount-via-sciserver" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'vignette-recount\', 1)"><img src="https://raw.githubusercontent.com/leekgroup/recount-website/master/sciserver_demo/demo1.png" align="center"/></a>'))
    ),
    tabPanel('Contribute your data',
        h3('Contribute your data to recount2!'),
        HTML('If you are interested in contributing your human RNA-seq data sequenced on the Illumina platform to recount2, please check how to do so at <a href="https://github.com/leekgroup/recount-contributions">recount-contributions</a>. Thank you!')
    ),
    tags$hr(),
    h3('Data license'),
    p('The data in ', strong('recount2'), HTML(' is licensed under CC BY 4.0. The legal text can be found <a href="LICENSE.txt">here</a>.')),
    tags$hr(),
    h3('Acknowledgements'),
    p('This research was supported by NIH R01 GM105705. LCT was supported by Consejo Nacional de Ciencia y Tecnología México 351535. LCT and AEJ were supported by NIH 1R21MH109956-01. Amazon Web Services experiments were supported by AWS in Education research grants. Storage costs on S3 for TCGA runs were partially covered by a grant from Seven Bridges Genomics for use of the Cancer Genomics Cloud.'),
	p(strong('recount2'), ' is hosted on ', HTML('<a href="http://www.sciserver.org" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'sciserver\', 1)">SciServer</a>,'), 'a collaborative research environment for large-scale data-driven science. It is being developed at, and administered by, the ', HTML('<a href="http://idies.jhu.edu/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'idies\', 1)">Institute for Data Intensive Engineering and Science (IDIES)</a>'), ' at Johns Hopkins University. SciServer is funded by the National Science Foundation Award ACI-1261715.'),
    p(HTML('<a href="http://idies.jhu.edu/" onclick="ga(\'send\', \'event\', \'click\', \'link\', \'idies\', 1)"><img src="http://idies.jhu.edu/wp-content/themes/idies/assets/img/idieslogo.png" align="right"/></a>')),
    tags$br(),
    tags$br(),
    tags$br(),
    tags$br()
))